miner/worker.go

```diff
@@ -937,8 +939,34 @@ type generateParams struct {
 	beaconRoot  *common.Hash      // The beacon root (cancun field).
 	noTxs       bool              // Flag whether an empty block without any transaction is expected
 
-	txs      types.Transactions // Deposit transactions to include at the start of the block
-	gasLimit *uint64            // Optional gas limit override
+	txs       types.Transactions // Deposit transactions to include at the start of the block
+	gasLimit  *uint64            // Optional gas limit override
+	interrupt *atomic.Int32      // Optional interruption signal to pass down to worker.generateWork
+}
+
+// validateParams validates the given parameters.
+// It currently checks that the parent block is known and that the timestamp is valid,
+// i.e., after the parent block's timestamp.
+func (w *worker) validateParams(genParams *generateParams) error {
```

Comment:
This is largely copied from the beginning of `prepareWork`.

An alternative would be to extract this part out of `prepareWork` and reuse it there, but then we would also need to return the `parent` and `timestamp`, which I found cumbersome, so opted for the copy.